### PR TITLE
Fix issue_12555. limit((3**x + 2 * x**10) / (x**10 + E**x), x, -oo) gives 0 instead of 2

### LINKED ISSUE #12555: limit((3**x + 2 * x**10) / (x**10 + E**x), x, -oo) gives 0 instead of 2
The following code

```
x = Symbol('x', real = True)
expr = (3**x + 2 * x**10) / (x**10 + E**x)
print(limit(expr, x, -oo))

```
returns `0` instead of `2`.

### PR DESCRIPTION
**Pull request by maurogaravello**
The wrong behavior is due to the fact that leading terms of numerator and denominator was calculated with a mistake. At some point the substitution `x = 1/u` was performed and then the leading terms of numerator and denominator were calculated comparing the order of the addends. In the case of the numerator we have `O(3**(1/u))` and `O(u**(-10))`.
When comparing the previous order, sympy assumes that `u` tends to `0`, but positive. Instead in this situation `u` should be negative.

Fixes #12555.

### CODE REVIEW

File: sympy/series/limits.py [hunk 1]
```diff
@@ -162,8 +162,12 @@ def doit(self, **hints):
                                         for m in Mul.make_args(a))
                                     for a in Add.make_args(w)))
                 if all(ok(w) for w in e.as_numer_denom()):
-                    u = Dummy(positive=(z0 is S.Infinity))
-                    inve = e.subs(z, 1/u)
+                    if z0 is S.NegativeInfinity:
+                        u = Dummy(positive=True)
+                        inve = e.subs(z, -1/u)
+                    else:
+                        u = Dummy(positive=(z0 is S.Infinity))
```
Comment (jksuom) [inline sympy/series/limits.py]: Is the condition `z0 is S.Infinity` necessary? We know that `abs(z0) == oo`. It seems that if `z0` is not `-oo`, then it should be `oo`.

Comment (maurogaravello) [inline sympy/series/limits.py]: I am not sure. If the limit is in the real field, then it is not necessary.
If the limit is in the complex field, probably it is better to have such condition.

**Comment by maurogaravello** [comment]
What is the check that failed?

File: sympy/series/limits.py (hunk 1 above)
Comment (jksuom) [inline sympy/series/limits.py]: I think that limit is currently taken on the real line alone. This is implicit in the the docstring

    z0 can be any expression, including oo and -oo.

    For dir="+" (default) it calculates the limit from the right
    (z->z0+) and for dir="-" the limit from the left (z->z0-).  For infinite
    z0 (oo or -oo), the dir argument is determined from the direction
    of the infinity (i.e., dir="-" for oo).

even though 'any expression' would seem to allow complex values.

**Comment by jksuom** [comment]
That is a job timed out, nothing serious.

File: sympy/series/limits.py (hunk 1 above)
Comment (maurogaravello) [inline sympy/series/limits.py]: In such a case, I modify the code.

File: sympy/series/limits.py [hunk 2]
```diff
@@ -162,8 +162,11 @@ def doit(self, **hints):
...
                 if all(ok(w) for w in e.as_numer_denom()):
-                    u = Dummy(positive=(z0 is S.Infinity))
-                    inve = e.subs(z, 1/u)
+                    u = Dummy(positive=True)
+                    if z0 is S.NegativeInfinity:
+                        inve = e.subs(z, -1/u)
+                    else:
+                        inve = e.subs(z, 1/u)
                     r = limit(inve.as_leading_term(u), u,
                               S.Zero, "+" if z0 is S.Infinity else "-")
```
Comment (jksuom) [inline sympy/series/limits.py]: `dir` should probably be always `+` here now.

Comment (maurogaravello) [inline sympy/series/limits.py]: You right!

**Comment by jksuom** [comment]
This looks good. Thanks.

**Comment by maurogaravello** [comment]
Also thanks for your suggestions.

**Pull request merged**